### ISSUE TITLE
[BUG] Campaigns menu link in dashboard returns 404

### ISSUE THREAD

**Issue by BenSturmfels** [houdiniproject/houdini#1078]
On the main branch, when logged in as the user created during the initial `bin/rails houdini:nonprofit:create --super-admin --no-confirm-admin ...` if I click the "Campaigns" link in the left-hand-side menu, I get a 404. Console says:

```
method=GET path=/ny/brooklyn/my-org/campaigns format=html controller=CampaignsController action=index status=404 error='ActionController::RoutingError: Not Found' duration=144.79 view=0.00 db=6.94 time=447613.51 exception=["ActionController::RoutingError", "Not Found"] exception_object=Not Found
  
ActionController::RoutingError (Not Found):
  
app/controllers/campaigns_controller.rb:134:in `check_nonprofit_status'
app/controllers/concerns/controllers/locale.rb:22:in `switch_locale'
```

Line 134 is:

`raise ActionController::RoutingError, 'Not Found' if !current_role?(:super_admin) && !current_nonprofit.published`

This raises a few questions:

1. Shouldn't my user by be a "super_admin" given it was created with `--super-admin`? If I look up the `roles` table in the database, this user is listed as a `nonprofit_admin`.
2. Should nonprofits created by the CLI perhaps be published by default?
3. What's the intended workflow for publishing a nonprofit? I can manually get into the `nonprofits` table set mark `published = true`, but that's probably not ideal. Can it be done via the UI? Do I have to be a super admin? If I manually change my role to `super_admin` in the database, I now only get three menu items on the left of the dashboard: "Profile", "Settings" and "Logout". The "Settings" link fails with `No route matches {:action=>"show", :controller=>"nonprofits", :id=>nil}, missing required keys: [:id]`.

**Comment by wwahammy** [comment]
@BenSturmfels Good questions:

1. You should have been a super admin when created that way, there was a bug preventing it from happening which should be fixed by #1082. Also, the users role as a super_admin is different from their role as a nonprofit_admin. So you'd need both roles to see stuff related to the a nonprofit. I suspect super_admin does not work without nonprofit_admin also but I'm not sure; I doubt it's ever been tested. 
2. Yes a nonprofit should be published by default. I'll make a fix for that.
3. A super admin can control whether a nonprofit is published by going to the nonprofits page and setting published in the top right. Also, please file a separate bug for that Settings link issue.

**Comment by BenSturmfels** [comment]
Thanks @wwahammy, that's working well for me on main. I don't think my last point about the settings link is worth bothering with given it's an inconsistent state (super admin role without nonprofit admin role). Closing.
